### ISSUE TITLE
[FEATURE] Allow Shizuku to pair with app

### ISSUE THREAD

**Issue by rgrimjr40** [Smooth-E/wireless-adb-switch#31]
> Describe the feature you would like to be added to WADBS. Make sure that the description is clear and detailed enough for a reader (developer or a new contributor) to understand.

- [ ] This feature was not requested before (both as a feature request and as a bug report that such feature would fix)
- [X ] I agree to keep the discussion polite and on topic

Would it be possible to allow Shizuku to recieve the pairing code? I have an LG v60. There is no option for wireleas debugging in developer options. I heard all LG devices are this way. Therefore, everytime I re-start my phone I have to use my PC to start Shizuku which is a hassle for me. You app works on my phone. Shizuku just will not connect with it. I asked in Shizuku github for the dev to add an option to allow a pairing code. I never got a response from him. Is there something that can be done via code to allow your app to make the connection?

**Comment by Smooth-E** [comment]
> - [ ] This feature was not requested before (both as a feature request and as a bug report that such feature would fix)
> - [X ] I agree to keep the discussion polite and on topic

Here is how you can check checkboxes in GitHub Markdown (just so you know):
```md
- [ ] Unchecked box
- [X] Checked box
```

---

 > Would it be possible to allow Shizuku to recieve the pairing code? 

Adding the functionality for displaying the pairing code was already requested in #30. I will work on this when I have time for this project. Can't provide an estimate for when exactly.

>  There is no option for wireleas debugging in developer options. I heard all LG devices are this way. Therefore, everytime I re-start my phone I have to use my PC to start Shizuku which is a hassle for me. You app works on my phone.

Does this mean that you are using WADBS in Shizuku mode and in root mode? In other words, does your routine looks like "enable Shizuku from a PC, then enable wireless debugging with WADBS in Shizuku mode"?

> You app works on my phone. Shizuku just will not connect with it.

What do you mean by "Shizuku just will not connect with it"?

> I asked in Shizuku github for the dev to add an option to allow a pairing code. I never got a response from him.

Please, link this issue (or the discussion somewhere else).

**Comment by rgrimjr40** [comment]
Thanks for your response. I start Shizuku by connecting my phone to my PC and running the ADB command provided by Shizuku. This allows me to run your app and other Shizuku apps. I do not have root. 

Once I have wireless debugging up and running from doing the above if I click "Pairing" below "Start via wireless debugging" in Shizuku it will not connect to wireless adb. It says Shizuku is searching and it never pairs.

I was hoping to be able to pair things this way that way I would only need the computer once to initially start Shizuku via ADB debugging command. Then, after it is paired I could select enable on boot in your app to keep Shizuku connected.

**Comment by Smooth-E** [comment]
> Once I have wireless debugging up and running from doing the above if I click "Pairing" below "Start via wireless debugging" in Shizuku it will not connect to wireless adb. It says Shizuku is searching and it never pairs.

If you have started Shizuku from the PC, there is no need to connect via wireless debugging. Shizuku can be activated (started) via either ADB through wire or via wireless ADB. If you have started it using one of the methods, . there is no point in using another method. 

Privilege level does not depend on the method you have activated Shizuku with. ADB (wired and wireless) methods of activation provide the "adb" level of privileges to apps that use Shizuku's API. Activating with root provides the "root" level of privileges to apps that use Shizuku's API.

> I was hoping to be able to pair things this way that way I would only need the computer once to initially start Shizuku via ADB debugging command. Then, after it is paired I could connect Shizuku using your app. Finally, enable on boot in your app to keep Shizuku connected.

Pairing via wireless debugging will reset after a reboot (same as with wired ADB method). Here is a [quote from Shizuku's "User manual" page](https://shizuku.rikka.app/guide/setup/#start-via-wireless-debugging):

> Starting with wireless debugging works on Android 11 or above. This startup method does not require a connection to a computer. Due to system limitations, the startup steps need to be performed again after each reboot.

Since you do not have root and your device is lacking a Developer Settings entry for wireless debugging, here is how you can alternatively benefit from WADBS:

- Use the wireless debugging feature (after activating through wire)
- View your device's IP address in the local network
- View the pairing code (will be added later) for wireless debugging and use it to connect **to your PC** quicker

Note that WADBS is oriented more on developers that need quick and easy way to start using wireless ADB when they are getting to they workplace. WADBS was not intended to provide workarounds for other apps and modifications. However, it is nice to see that it helps people to get previously locked features on their devices.

**Comment by rgrimjr40** [comment]
I was hoping to find a way to get Shizuku to connect via wireless debugging. I thought if somehow I coukd get your app to connect to Shizuku wireless ADB would survive reboot. Sorry I did not think the process out very well.

I am glad I got to talk to you and that I found your app. You are  right. I can use WADBS to connect the PC to my phone faster. Sometimes it can be a pain on this phone doing it manually. I appreciate your help greatly. Thank you for taking the time to help me out. Once again sorry for the confusion 😉
